File: .web3scripts/ecrecover.py
from web3.auto import w3
from eth_account.messages import defunct_hash_message
import sys


def recover_address(message, signature):
    message_hash = defunct_hash_message(text=message)
    address = w3.eth.account.recoverHash(message_hash, signature=signature)
    return address


address = recover_address(sys.argv[1],sys.argv[2])
print(address)

File: .web3scripts/ecrecover.rb
require 'eth'
message = ARGV[0]
signature = ARGV[1]

#p "0:#{ARGV[0]}"
#p "1:#{ARGV[1]}"
#p "2:#{ARGV[2]}"

public_key = Eth::Signature.personal_recover(message,signature)

address = Eth::Util.public_key_to_address(public_key)
p address
